services/controler.js

```diff
@@ -1,11 +1,86 @@
-import logger from '../services/logger.js'
-
-const images = (req,res)=>{
-    let reqURI = req.originalUrl;
-    
-    logger.debug('reqURI: ' + reqURI)
-    
-    res.status(200).json({message:  `You requested the following URI: ${reqURI}`})
+import establishRedis from '../services/redis.js';
+import logger from '../services/logger.js';
+
+// List of available target sizes
+const targetSizes = [32, 64, 128, 256, 512, 1024, 2048, 4096, 8192];
+
+// Find the nearest available resolution given the requested size
+function getNearestResolution(requestedSize) {
+    const reqNum = parseInt(requestedSize, 10);
+    if (isNaN(reqNum)) return 'original';
+    const nearest = targetSizes.reduce((prev, curr) =>
+        Math.abs(curr - reqNum) < Math.abs(prev - reqNum) ? curr : prev
+    );
+    return `${nearest}x${nearest}`;
+}
+
+// Retrieve image from Redis and return binary Buffer
+async function getImage({ name, resolution: size }) {
+    const redisConnection = await establishRedis();
+    const redisKey = `image:${name}`;
+    // If size is provided, find the nearest resolution; otherwise, use original
+    const resolution = size ? getNearestResolution(size) : 'original';
+
+    try {
+        await redisConnection.connect();
+        const base64Image = await redisConnection.hGet(redisKey, resolution);
+        if (!base64Image) {
+            logger.error(`No image found for ${redisKey} at resolution ${resolution}`);
+            return null;
+        }
+        return Buffer.from(base64Image, 'base64');
+    } catch (err) {
+        logger.error(`Error retrieving image ${name} at resolution ${resolution}: ${err}`);
+        return null;
+    } finally {
+        await redisConnection.disconnect();
+    }
 }
 
-export default images
\ No newline at end of file
+// Parse image data from the URL
+const getImageData = (uri) => {
+    // Remove any leading slash
+    const trimmed = uri.startsWith('/') ? uri.slice(1) : uri;
```

Comment:
According to documentation the leading slash is always there.

```suggestion
    const trimmed = uri.slice(1);
```